core/modules/ckeditor/js/plugins/backdroplink/plugin.js

```diff
@@ -70,50 +70,89 @@ CKEDITOR.plugins.add('backdroplink', {
+          selectedText = linkElement.$.text;
+        }
 
-        // Set existing values based on selected element.
+        // Set existing attribute values based on selected element.
         var existingValues = {};
         if (linkElement && linkElement.$) {
           existingValues = parseAttributes(editor, linkElement);
+
+          // Update the displayed link text
+          existingValues.linktext = selectedText;
         }
         // Or, if an image widget is focused, we're editing a link wrapping
         // an image widget.
         else if (focusedImageWidget && focusedImageWidget.data.link) {
           existingValues = CKEDITOR.tools.clone(focusedImageWidget.data.link);
         }
+        // Or, if selected element is not an existing link or an image.
+        else {
+          existingValues.linktext = selectedText;
+        }
 
         // Prepare a save callback to be used upon saving the dialog.
         var saveCallback = function (returnValues) {
           editor.fire('saveSnapshot');
-
           // Ignore a disabled target attribute.
           if (returnValues.attributes.target === 0) {
             delete returnValues.attributes.target;
           }
-
           // If an image widget is focused, we're not editing an independent
           // link, but we're wrapping an image widget in a link.
           if (focusedImageWidget) {
             focusedImageWidget.setData('link', CKEDITOR.tools.extend(returnValues.attributes, focusedImageWidget.data.link));
             editor.fire('saveSnapshot');
             return;
           }
-
-          // Create a new link element if needed.
-          if (!linkElement && returnValues.attributes.href) {
+          // If not an image, replace text of link with new text.
+          else {
+            // Get the current selection.
             var selection = editor.getSelection();
+            // Get the text of the current selection.
+            var oldtext = selection.getSelectedText();
+            // And the replacement text.
+            var newtext = returnValues.attributes.linktext;
+            // Get the range of the selection.
             var range = selection.getRanges(1)[0];
 
+            // If the selection is a link, replace the text
+            var element = selection.getStartElement();
+            this.element = element;
+            if(element.$.nodeName === "A") {
+              this.element.setText(newtext);
+            }
+          }
+
+          // Create a new link element if needed.
+          if (!linkElement && returnValues.attributes.href) {
             // Use link URL as text with a collapsed cursor.
             if (range.collapsed) {
-              // Shorten mailto URLs to just the email address.
-              var text = new CKEDITOR.dom.text(returnValues.attributes.href.replace(/^mailto:/, ''), editor.document);
+
+              if(newtext != ""){
+                var text = new CKEDITOR.dom.text( newtext );
```

Comment:
minor coding standard here, remove space in parenthesis

Reply:
Yes, thanks.